### PR TITLE
Make Culprit Finder smarter

### PR DESCRIPTION
**Pull request by meteorcloudy**
Culprit finder can now find out what default value to use, if some environment variables are not set.
- `TASK_NAME` or `PLATFORM_NAME`: If not set, bisect for all tasks of the specified project
- `GOOD_BAZEL_COMMIT`: If not set, use the last green bazel commit in downstream pipeline
- `BAD_BAZEL_COMMIT`: If not set, use the latest Bazel commit

After this, to do a bisect, you only have to set `PROJECT_NAME`

FYI @hlopko @vladmos @buchgr

### CODE REVIEW

**Comment by meteorcloudy** [comment]
Tested at https://buildkite.com/bazel/pcloudy-test/builds/64#_

File: buildkite/culprit_finder.py [hunk 1]
```diff
@@ -174,10 +183,25 @@ def main(argv=None):
...
             # For old config file, we can still set PLATFORM_NAME as task name.
-            task_name = os.environ.get("PLATFORM_NAME") or os.environ["TASK_NAME"]
-            good_bazel_commit = os.environ["GOOD_BAZEL_COMMIT"]
-            bad_bazel_commit = os.environ["BAD_BAZEL_COMMIT"]
+            if "PLATFORM_NAME" in os.environ or "TASK_NAME" in os.environ:
+                tasks = [os.environ.get("PLATFORM_NAME") or os.environ["TASK_NAME"]]
+            else:
+                tasks = get_tasks(project_name)
+
+            if "GOOD_BAZEL_COMMIT" in os.environ:
```
Comment (fweikert) [inline buildkite/culprit_finder.py]: Nit: I'd prefer

good_bazel_commit = os.environ.get("GOOD_BAZEL_COMMIT")
if not good_bazel_commit:
  # If GOOD....

File: buildkite/culprit_finder.py [hunk 2]
```diff
@@ -174,10 +183,25 @@ def main(argv=None):
     if args.subparsers_name == "culprit_finder":
         try:
             project_name = os.environ["PROJECT_NAME"]
+
             # For old config file, we can still set PLATFORM_NAME as task name.
-            task_name = os.environ.get("PLATFORM_NAME") or os.environ["TASK_NAME"]
-            good_bazel_commit = os.environ["GOOD_BAZEL_COMMIT"]
-            bad_bazel_commit = os.environ["BAD_BAZEL_COMMIT"]
+            if "PLATFORM_NAME" in os.environ or "TASK_NAME" in os.environ:
+                tasks = [os.environ.get("PLATFORM_NAME") or os.environ["TASK_NAME"]]
```
Comment (fweikert) [inline buildkite/culprit_finder.py]: Technically this code could crash if PLATFORM_NAME was set to an empty string and there was no TASK_NAME. But it's very unlikely.

Comment (meteorcloudy) [reply buildkite/culprit_finder.py]: Yeah, refactor this as well. Thanks!

File: buildkite/culprit_finder.py (hunk 1 above)
Comment (meteorcloudy) [reply buildkite/culprit_finder.py]: Done.

**Pull request merged**